components/TextFields/src/MDCTextField.m

```diff
@@ -106,6 +111,72 @@ - (void)commonMDCTextFieldInitialization {
                       object:self];
 }
 
+#pragma mark - Underline View Implementation
+
+- (void)setupUnderlineConstraints {
+  NSLayoutConstraint *underlineLeading = [NSLayoutConstraint constraintWithItem:self.underline
+                               attribute:NSLayoutAttributeLeading
+                               relatedBy:NSLayoutRelationEqual
+                                  toItem:self
+                               attribute:NSLayoutAttributeLeading
+                              multiplier:1
+                                                                       constant:0];
+  underlineLeading.priority = UILayoutPriorityDefaultLow;
+  underlineLeading.active = YES;
+
+  NSLayoutConstraint *underlineTrailing = [NSLayoutConstraint constraintWithItem:self.underline
+                               attribute:NSLayoutAttributeTrailing
+                               relatedBy:NSLayoutRelationEqual
+                                  toItem:self
+                               attribute:NSLayoutAttributeTrailing
+                              multiplier:1
+                                                                        constant:0];
+  underlineTrailing.priority = UILayoutPriorityDefaultLow;
+  underlineTrailing.active = YES;
+
+  CGFloat estimatedTextHeight = MDCCeil(self.font.lineHeight * 2.f) / 2.f;
+  _underlineY =
+  [NSLayoutConstraint constraintWithItem:self.underline
+                               attribute:NSLayoutAttributeCenterY
+                               relatedBy:NSLayoutRelationEqual
+                                  toItem:self
+                               attribute:NSLayoutAttributeTop
+                              multiplier:1
+                                constant:[self textInsets].top + estimatedTextHeight +
+   MDCTextInputHalfPadding];
+  _underlineY.priority = UILayoutPriorityDefaultLow;
+  _underlineY.active = YES;
+}
+
+- (CGFloat)underlineYConstant {
+  CGFloat estimatedTextHeight = MDCCeil(self.font.lineHeight * 2.f) / 2.f;
```

Comment:
This same estimation is in intrinsicContentSize and above in setupUnderlineConstraints. Consider turning it into a method.